runbooks/mcp-to-permissioned-fps.md

```diff
@@ -0,0 +1,382 @@
+- `deployments.json` is the list of addresses deployed
+- `standard-addresses.json` is the list of standard implementation addresses used
+- `transactions.json` is the list of transactions that were executed
+
+**Deployment cost**
+
+We recommend providing the deployer account with 1 ETH on Mainnet and 10 ETH on Sepolia to minimize the chance of a failed deployment. Actual deployment cost should be significantly less than these recommended values.
+
+- Estimated Gas: `12,000,000`
+- Mainnet Cost: `~0.15 ETH @ ~13gwei`
+- Sepolia Cost: `~3 ETH @ ~250gwei`
+
+## Executing the deployment
+
+1. Create a working directory
+    
+    ```bash
+    mkdir upgrade-dir
+    cd upgrade-dir
+    ```
+    
+2. Copy your deploy config and deployment addresses JSON files into the working directory
+    
+    > [!IMPORTANT]
+    > Please make sure that your deploy config JSON is in the standard config format used by the official OP Stack deployment script and includes all required configuration values as of `op-contracts/v1.3.0`.
+    
+3. Make sure that your files are named correctly, deploy config should be named `deploy_config.json` your deployment addresses file should be named `deployments.json`
+    
+    > [!NOTE]
+    > Standardized file names reduce the chance of errors when running this runbook.
+        
+4. Create a folder called `outputs`
+    
+    ```bash
+    mkdir outputs
+    ```
+    
+5. Create a file called `.env`
+        
+    ```bash
+    touch .env
+    ```
+
+6. Add the following to `.env` and fill out all required variables
+    
+    ```bash
+    ##############################################
+    #               ↓  Required  ↓               #
+    ##############################################
+
+    # Can be "mainnet" or "sepolia"
+    NETWORK=
+
+    # Etherscan API key used to verify contract bytecode
+    ETHERSCAN_API_KEY=
+
+    # RPC URL for the L1 network that matches $NETWORK
+    ETH_RPC_URL=
+
+    # Private key used to deploy the new contracts for this upgrade
+    PRIVATE_KEY=
+
+    # Check if required files and folders exist
+    if [ ! -f "./deploy_config.json" ]; then
+        echo "Error: deploy_config.json not found"
+    fi
+    if [ ! -f "./deployments.json" ]; then
+        echo "Error: deployments.json not found"
+    fi
+    if [ ! -d "./outputs" ]; then
+        echo "Error: outputs folder not found"
+    fi
+    ```
+    
+7. Load the `.env` file into your environment
+    
+    If you get an error when running this command, make sure that your input files are properly named and that you have created the `outputs` folder.
+
+    ```bash
+    source .env
+    ```
+    
+8. Run the deployment process
+    
+    ```bash
+    docker run -t \
+        --env-file .env \
+        -v ./deploy_config:/deploy_config.json \
+        -v ./deployments.json:/deployments.json \
+        -v ./outputs:/outputs \
+        kfoplabs/upgrade-v1.3.0-v1.6.0-permissioned:latest \
```

Comment:
Should we update this to be a fixed hash rather than latest?

Reply:
@smartcontracts do we have a finalized version to reference here?

Reply:
sha256:dfb661ac34e997694eb0aaa2536515992cc0a3d663bfe3adbac0deb7aa1dd159